### PR TITLE
bootloader: make bootloader flash area reserve optional

### PR DESCRIPTION
**Pull request by dron0gus**
also increase config flash area to 8K

### CODE REVIEW

File: firmware/boards/f1_rev2/wideband_layout.ld [hunk 1]
```diff
@@ -4,9 +4,10 @@
 
 MEMORY
 {
-    /* 8k reserved for bootloader */
-    appflash (rx) : org = 0x08000000 + 8k, len = 128k - 1k - 8k
-    configflash (rx) : org = 0x08000000 + 128k - 1k, len = 1k
+    /* empty section to keep linker happy */
+    blflash (rx) : org = 0x08000000, len = 0
+    appflash (rx) : org = DEFINED(USE_BOOTLOADER) ? (0x08000000 + 8k) : 0x08000000, len = DEFINED(USE_BOOTLOADER) ? (128k - 8k - 8k) : (128k - 8k)
+    configflash (rx) : org = 0x08000000 + 128k - 8k, len = 8k
```
Comment (mck1117) [inline firmware/boards/f1_rev2/wideband_layout.ld:10]: why is the config so large?

**Comment by rusefillc** [comment]
@mck1117 what would be acceptable size for you?

**Comment by mck1117** [comment]
My question is not exactly why it's 8k, but why it needs to be bigger than the existing 1k

**Comment by rusefillc** [comment]
CHibiOS fancy HAL_USE_EFL and MFSConfig encapsulate persistance redundancy, we would definitely need at least two pages i.e. at least 2K

8K here is an overkill of twice 4K but maybe we can lower the bar a bit for the sake of velocity?

**Comment by dron0gus** [comment]
More flash for small config - more times you can save it into MFS bankX just by appending it without need to erase.

**Pull request merged**